File: 258-add-digits/add-digits.cpp
class Solution {
public:

    int ans(int n){
        if(n == 0)return 0 ;
        return n%10 + ans(n / 10);
    }


    int addDigits(int num) {
        
        int a = ans(num) ;

        while(a >= 10 ){
            a = ans(a);
        }

        return a ;

    }
};

File: 258-add-digits/add-digits.java
class Solution {
    public int sumOfNumbers(int n, int sum) {

        if( n == 0 && sum < 10)
        {
            return sum ;
        }

        else if(n == 0 && sum >= 10){
            n = sum ;
            return sumOfNumbers(n , 0);
        }

        else{
            return sumOfNumbers(n/10 , sum += n%10);
        }

    }
    public int addDigits(int num) {
        return sumOfNumbers(num , 0 );
    }
}